src/pages/main/components/selected-dataset/build-flow/components/secondary-ds-item.tsx

```diff
@@ -20,6 +21,13 @@ interface IDatasetsListItemProps {
 export const SecondaryDsItem: FC<IDatasetsListItemProps> = observer(
   ({ item, level = 1, onSelect, selectedItem }) => {
     const params = useParams()
+    const { droppedDs } = defaultsStore
+
+    const isAbleToBeDropped = useMemo(() => {
+      return droppedDs.includes(item.name)
+    }, [droppedDs, item.name])
+
+    const [isDsHovered, setIsDsHovered] = useState(false)
```

Comment:
You shouldn't use state here, please use css styling for it

Reply:
In accordance with the task description, I have to show the bucket icon when a user hovers or clicks on the secondary ds. It's not about styling.

Reply:
By using css styles for bucket, you can change it `display` value from` initial `to `none`, depending on hovering parent element, and reduce amount of rerendering by react. As it is standard browser ability it is preferable to use it, instead of conditional rendering. + if element is active, you can just add style in JSX for bucket `display: initial !important`

Reply:
I don't see how task description conflict with this approach

Reply:
ok, didn't see the point